lms/models/file.py

```diff
@@ -0,0 +1,47 @@
+import sqlalchemy as sa
+
+from lms.db import BASE
+from lms.models._mixins import BulkUpsertMixin, CreatedUpdatedMixin
+
+
+class File(CreatedUpdatedMixin, BulkUpsertMixin, BASE):
+    """A record of files we've seen in LMS's."""
+
+    __tablename__ = "file"
+    __table_args__ = (
+        sa.UniqueConstraint("application_instance_id", "lms_id", "type", "course_id"),
```

Comment:
I think this index ordering will support the query patterns we are likely to have:

 * We'll always search by `application_instance_id`
 * We're mostly interested in single files, so the ID seems very likely
 * The type seems like the next most likely part to be in any query
 * Only the course part is probably likely to be dropped

As long as we use the columns from the start, we should be able to leverage this for our queries